tests/FieldsBuilderTest.php

```diff
@@ -197,6 +200,46 @@ public function test($typeHintInDocBlock)
         $this->assertInstanceOf(StringType::class, $query->getType()->getWrappedType());
     }
 
+    /**
+     * Tests that the fields builder will fail when a parameter is missing it's generic docblock
+     * definition, when required - an array, for instance, or could be a collection (List types)
+     *
+     * @requires PHP >= 8.0
+     */
+    public function testTypeMissingForPropertyPromotionWithoutGenericDoc(): void
+    {
+        $fieldsBuilder = $this->buildFieldsBuilder();
+
+        $this->expectException(CannotMapTypeException::class);
+
+        $fieldsBuilder->getInputFields(
+            PropertyPromotionInputTypeWithoutGenericDoc::class,
+            'PropertyPromotionInputTypeWithoutGenericDocInput',
+        );
+    }
+
+    /**
+     * Tests that the fields builder will properly build an input type using property promotion
+     * with the generic docblock defined on the constructor and not the property directly.
+     *
+     * @requires PHP >= 8.0
```

Comment:
```suggestion
     * @requires PHP >= 8.0
```

It is not necessary, since library supports only 8.1+

Reply:
Agreed, was following similar fixture formats.  We probably need to clean all that up now that 8.1 is min support.

Reply:
Will submit all this under another PR